referenceframe/frame.go

```diff
@@ -630,3 +665,83 @@ func PoseToInputs(p spatial.Pose) []Input {
 		p.Orientation().OrientationVectorRadians().Theta,
 	})
 }
+
+// Determine whether two frames are (nearly) identical. For now, we only support implementers of
+// the frame interface that are registered (see register.go). However, this is not automatic, so
+// if a new implementer is registered manually in register.go, its case should be added here.
+//
+// NOTE: for ease, this function only takes one epsilon parameter because we have yet
+// to see a case of quantity where we want accept different levels of floating point error.
+// If the time comes where we want a different allowance for limits, vectors, and geometries,
+// this function should be changed accordingly.
+func framesAlmostEqual(frame1, frame2 Frame, epsilon float64) (bool, error) {
+	switch {
+	case reflect.TypeOf(frame1) != reflect.TypeOf(frame2):
+		return false, nil
+	case frame1.Name() != frame2.Name():
+		return false, nil
+	case !limitsAlmostEqual(frame1.DoF(), frame2.DoF(), epsilon):
+		return false, nil
+	default:
+	}
+
+	switch f1 := frame1.(type) {
+	case *staticFrame:
+		f2 := frame2.(*staticFrame)
+		switch {
+		case !spatial.PoseAlmostEqual(f1.transform, f2.transform):
+			return false, nil
+		case f1.geometry == nil && f2.geometry == nil:
+			return true, nil
+		case !spatial.GeometriesAlmostEqual(f1.geometry, f2.geometry):
+			return false, nil
+		default:
+		}
+	case *rotationalFrame:
+		f2 := frame2.(*rotationalFrame)
+		if !spatial.R3VectorAlmostEqual(f1.rotAxis, f2.rotAxis, epsilon) {
+			return false, nil
+		}
+	case *translationalFrame:
+		f2 := frame2.(*translationalFrame)
+		switch {
+		case !spatial.R3VectorAlmostEqual(f1.transAxis, f2.transAxis, epsilon):
+			return false, nil
+		case f1.geometry == nil && f2.geometry == nil:
+			return true, nil
+		case !spatial.GeometriesAlmostEqual(f1.geometry, f2.geometry):
+			return false, nil
+		default:
+		}
+	case *tailGeometryStaticFrame:
+		f2 := frame2.(*tailGeometryStaticFrame)
+		switch {
+		case f1.staticFrame == nil:
+			return f2.staticFrame == nil, nil
+		case f2.staticFrame == nil:
+			return f1.staticFrame == nil, nil
+		default:
+			return framesAlmostEqual(f1.staticFrame, f2.staticFrame, epsilon)
+		}
+	case *SimpleModel:
```

Comment:
This is a `SimpleModel`:
```type SimpleModel struct {
	*baseFrame
	// OrdTransforms is the list of transforms ordered from end effector to base
	OrdTransforms []Frame
	modelConfig   *ModelConfigJSON
	poseCache     sync.Map
	lock          sync.RWMutex
}
```

Do we need to check `*baseFrame`, `modelConfig` or `poseCache`?

Reply:
adding check for `modelConfig`. `poseCache` is just procedural bookkeeping and can change from the same model to a later state